### DISCUSSION TITLE
Is there a possibility to deactivate NVDA for a single application and only use my own data to let NVDA speak

### DISCUSSION THREAD

**Discussion by SF-Podszuweit** [nvaccess/nvda#16096] [Q&A] [answered]
Hello Everyone, 

I have been testing our Software with NVDA and came to the conclusion, that it didn't really work as I expected. 
The Problem is, that our IDE (PowerBuilder), is not really supporting any accessibility features when using the Tab key or any field that is in the edit state. It works when hovering above fields not in edit state, but using [tab] to scroll through the fields fails by the design of PowerBuilder which puts any edit field in a new window class.

So I tried an other approach: I used the controllerClient to give NVDA the strings to read on my own. 
With this, I'm making my own string right now, which includes data and context. 
I'm then using the function "nvdaController_speakText(string text)" to let NVDA do the speaking. 

This works great, but NVDA keeps going and starts speaking the texts it would as usual. I would like to disable this.
I've tried the sleep mode, but this doesn't work as expected. 
I have "Alt" + "N" as the NVDA-Key, but with "Shift"+"S" it seems to restart NVDA. 

So my Question is: is there any way to tell NVDA to "ignore" a application? So it only "speaks" if triggered programmatically via "speakText"?

Is this best practice to give the wanted/correct audio output? 
Are there any suggestions on what to do or anything that I am missing out.  Any hint will be appreciated.

Best regards, 
Jan Malte

**Comment by XLTechie** [comment]
I think there may be some confusion here.
You say you have alt+n as your NVDA key, but that is impossible.

Alt+n may be what you use to start NVDA, but the actual NVDA key is going to be 
either insert, numpad insert, the capslock key, or all three, depending on what 
you selected in the welcome dialog.

Thus, to activate sleep mode, the key sequence is one of:

Capslock+shift+s

Insert+shift+s

Numpad insert+shift+s

**Reply by SF-Podszuweit** [reply]
Okay, I see and tested this quick, my bad. Is there any other advice on the other questions?

**Comment by lukaszgo1** [comment]
While it is really nice you're trying to make PowerBuilder accessible for screen reader users, the proposed approach is not really scalable. It will not work for Braille users, will make all familiar NVDA shortcuts such as `NVDA+Tab` not working etc. In general controller client is only usable to present simple text notifications and in games, where standard keyboard interactions paradigms are not applicable. It will really be better to improve accessibility of PowerBuilder's GUI, rather than trying to hack around the problem.

**Comment by Brian1Gaff** [comment]
Yes I'd discourage making the app self voicing merely because you will, at some point need access to explorer or other in built windows system and many self voicing programs end up being silent here as they are in effect in Windows. Take Newsbridge, a program I use a lot but the author went to self voicing and hence in Windows routines, there are either two voices, or a big lag while windows sorts out what the heck its meant to be doing before the snappy nvda voice catches up. so as your program needs to be snappy, it really does need to be using its input from the software and using this to make a screenreader talk. Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message -----

**Reply by SF-Podszuweit** [reply]
Hello,

Yes, I am aware that my way isn't best practice. But besides that, is there any way to disable NVDA for a specific program?
And for your info, PowerBuilder isn't programmed by us it's a Develop Enviroment we work with to code our own software, but we don't have any influence on how accessibility is handled.  

Jan Malte

**Reply by hwf1324** [reply, marked as answer]
See: https://www.nvaccess.org/files/nvda/documentation/developerGuide.html#toc26

**Reply by SF-Podszuweit** [reply]
Thank you very much. This is what I was looking for. 

Thanks to everyone else.
